Release blog for rippled 2.3.1 (#2963)
XRPLF/xrpl-dev-portal: blog/2025/rippled-2.3.1.md

---
category: 2025
date: 2025-01-30
seo:
    title: Introducing XRP Ledger version 2.3.1

>     description: rippled version 2.3.1 is now available. This version introduces new features and stability fixes.

Reviewer: ```suggestion
    description: rippled version 2.3.1 is now available. This version includes stability fixes.
```
nit

labels:
    - rippled Release Notes
markdown:
    editPage:
        hide: true
---
# Introducing XRP Ledger version 2.3.1

Version 2.3.1 of `rippled`, the reference server implementation of the XRP Ledger protocol, is now available. 

This release fixes an ongoing issue where normal peer traffic is charged incorrectly, causing peers to be disconnected inappropriately during normal operations. This affects network connectivity and the ability of some UNL validators to reach consensus with the others.

> ## Action Required

Reviewer: ```suggestion
## Action Recommended
```
I think servers stable with 2.3.0 are ok to remain

Date:   Wed Jan 29 09:26:27 2025 -0500

    Set version to 2.3.1
```

## Full Changelog

### Amendments and New Features

- None

### Bug Fixes and Updates

- Reduce peer charges for well-behaved peers. ([#5243](https://github.com/XRPLF/rippled/pull/5243))
  - Fix an erroneous high fee penalty that some peers could incur for sending older transactions.
  - Update the fees charged for imposing a load on the server.
  - Fix to prevent interal pseudo-transactions from relaying. Previously, pseudo-transcations received from a peer failed the signature check because they had no signature--even if they were requested using `TMGetObjectByHash`. This caused the peer to be charged for an invalid signature. This fix places pseudo-transactions only in the global cache (TransactionMaster). If the transaction is not part of a `TMTransactions` batch, the peer is charged an unwanted data fee. These fees won't be a problem in the normal course of operations, but should dissuade peers from sending many unwanted transactions.
  - Improve logging to specify the reason for fees charged to a peer.

## Credits

The following people contributed directly to this release:

- Ed Hennis <ed@ripple.com>

> - JoelKatz <DavidJoelSchwartz@GMail.com>

Reviewer 2: We've previously credited him as:

```suggestion
- David 'JoelKatz' Schwartz <DavidJoelSchwartz@GMail.com>
```
